src/IdentityServer/Config.cs

```diff
@@ -2,46 +2,43 @@
 // Licensed under the Apache License, Version 2.0. See LICENSE in the project root for license information.
 
 using System.Collections.Generic;
+using System.Linq;
 using IdentityServer4.Models;
 using Microsoft.Extensions.Configuration;
 
 namespace IdentityServer
 {
     public class Config
     {
-        private readonly List<Client> _clients = new List<Client>();
-
         private readonly IConfiguration _config;
 
         public Config(IConfiguration config)
         {
             _config = config;
 
-            var clients = new List<Client>();
             IConfigurationSection configClients = _config.GetSection("Clients");
-            foreach (var configClient in configClients.GetChildren())
+            var configClient = configClients.GetChildren().FirstOrDefault();
+
+            var siteUri = configClient.GetValue<string>("SiteUri").TrimEnd('/');
+            DynamicClientStore.Client = new Client
             {
-                var siteUri = configClient.GetValue<string>("SiteUri").TrimEnd('/');
-                this.Clients.Add(new Client
-                {
-                    ClientId = configClient.GetValue<string>("ClientId"),
-                    AllowedGrantTypes = { configClient.GetValue<string>("AllowedGrantTypes") },
-                    RedirectUris =
+                ClientId = configClient.GetValue<string>("ClientId"),
+                AllowedGrantTypes = { configClient.GetValue<string>("AllowedGrantTypes") },
+                RedirectUris =
```

Comment:
perhaps just load these from the config directly instead of from siteuri

Reply:
Updates made as per comment.
Commit: a095b51